processing/src/main/java/io/druid/segment/filter/ExtractionFilter.java

```diff
@@ -57,12 +55,13 @@ public ExtractionFilter(
     if (allDimVals != null) {
       for (int i = 0; i < allDimVals.size(); i++) {
         String dimVal = allDimVals.get(i);
-        if (value.equals(fn.apply(dimVal))) {
+        if (value.equals(Strings.nullToEmpty(fn.apply(dimVal)))) {
```

Comment:
This won't handle the case where `value` itself is null.  I expect that to generate an NPE actually.

Can we instead do a null check and convert empty to null?